### PR TITLE
Msm users frontend.go

### PR DESCRIPTION
**Pull request by manpreet85**
Inventory.go based on workloads.go

It is skeletal and would need more work before it can be usefule

### CODE REVIEW

File: internal/services/frontend/doc.go [hunk 1]
```diff
@@ -101,12 +101,37 @@ Updates the workload record the for workload matching workloadname according to
...
+GET - /api/racks/{rackid}
+Returns a single rack ID record.
+
+GET - /api/racks/rack-id/{blades}
+Get list of known blades in a rack.
+
+GET - /api/racks/rack-id/{TOR}
+Get Top of Racks details .
+
+GET - /api/racks/rack-id/{PDU}
```
Comment (Jim3Things) [inline internal/services/frontend/doc.go:116]: TOR and PDU are literals, not variables.  They shouldn't be in {..}

File: internal/services/frontend/inventory.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	// it there allows simple experimentation with just a browser.
+	//
+	// As a reminder,
+	//	 PUT is idempotent so translates to UPDATE in the CRUD methodolgy
+	//   POST is NOT idempotent so translates to CREATE in the CRUD methodolgy
+	//
+	routeInventory.HandleFunc("/{rackid}", handlerInventoryCreate).Methods("POST", "GET") // May be only GET
+	routeInventory.HandleFunc("/{rackid}", handlerInventoryRead).Methods("GET")
+	routeInventory.HandleFunc("/{rackid}", handlerInventoryUpdate).Methods("PUT", "GET")
+	routeInventory.HandleFunc("/{rackid}", handlerInventoryDelete).Methods("DELETE", "GET")
```
Comment (Jim3Things) [inline internal/services/frontend/inventory.go:60]: We only need to support the GET method for now.

File: internal/services/frontend/doc.go [hunk 3]
```diff
@@ -101,12 +101,37 @@ Updates the workload record the for workload matching workloadname according to
...
 operation code is invalid in some way.
 
+GET - /api/racks
+Get the list of all known racks.
+
+GET - /api/racks/{rackid}
+Returns a single rack ID record.
+
+GET - /api/racks/rack-id/{blades}
+Get list of known blades in a rack.
```
Comment (Jim3Things) [inline internal/services/frontend/doc.go:111]: We'll want

GET /api/racks/{rack-id}/blades 
to get the list of blades associated with that rack and

GET /api/racks/{rack-id}/blades/{blade-id}
to get details about a specific blade in a specific rack

File: internal/services/frontend/doc.go [hunk 4]
```diff
@@ -101,12 +101,37 @@ Updates the workload record the for workload matching workloadname according to
...
+
 
 
 TODO
+
 /api/logs
+
 /api/stepper
 /api/injector
 /api/inventory
```
Comment (Jim3Things) [inline internal/services/frontend/doc.go:133]: The code has all the inventory under /api/inventory, but the examples above don't.  We just need to resolve whether /inventory or /blades is the root term

**Comment by manpreet85** [comment]
> The code has all the inventory under /api/inventory, but the examples above don't.  We just need to resolve whether /inventory or /blades is the root term

**Comment by manpreet85** [comment]
Can you please advise what is root term and how it impacts the code?

**Pull request merged**